adobe/spectrum-web-components
# refactor: use "autoUpdate" to place/move/close an overlay

## Description
- update floating-ui dependency
- wrap the second call to `computePosition` in `autoUpdate` so that we get ancestor size/location reactivity
  - close the overlay on subsequent calls when not a "modal"
- add tests
- add demos

## How has this been tested?

    1. Go [here](https://overlay-close--spectrum-web-components.netlify.app/storybook/?path=/story/overlay--click-content-closed-on-scroll)
    2. See that the page scroll is not at 0
    3. See that the overlay has closed
    4. Open the overlay again
    5. Scroll the page
    6. See the overlay close

## Types of changes

## Checklist
